ROADMAP.md

```diff
@@ -0,0 +1,69 @@
+
+## Data retention policies
+The length of time that data will be retained in Strooms stream store can be defined by creating data retention rules. These rules match streams based on their meta data and will automatically delete data once the retention period associated with the rule is exceeded.
+
+## Dashboard linking
+Links can be created in dashboards to jump to other dashboards or other external sites that provide additional contextual information.
+
+## Annotations
+Search results in dashboards can be annotated to provide status and notes relating to the result item, e.g. an event. These annotations can later be searched to see which events have annotations associated with them.
+
+## Search API
+The search system used by Dashboards can be used via a restful API. This provides access to data stored in search indices (including the ability to extract data) and statistics stores. The data fetched via the search API can be received and processed via an external system.
+
+## Kafka appender and filter
+New pipeline elements for writing XML or text data to a Kafka topic. This provides more options for using Stroom's data in other systems.
+
+# v7.0
+
+## Authorisation enhancements
+The Stroom authorisation system is split out into a separate service and provides integration with external authorisation mechanisms.
+
+## Reference data storage
+Reference data uses a memory mapped disk based store rather than direct memory to reduce the memory overhead associated with storing reference data. Reference data is also enhanced to cope with changes (additions and removals) of state information rather than always relying on complete snapshots.
+
+## Proxy processing
+Stroom proxy is capable of pipeline processing in the same way as a full Stroom application. Pipeline configuration content can be pushed to proxies so that they can perform local processing prior to sending data to Stroom. 
+
+## Multiple input sources
+Stroom is capable of processing data from a Kafka topic, HDFS, the local file system, HTTP POST in addition to the included stream store.
+
+In addition to processing data from multiple sources Stroom now has improved support for writing to various destinations.
+
+## Improved field extraction
+Enhancements to data splitter and associated UI to make the process of extracting field data from raw content much easier.
+
+## Kafka backed processing
+Stroom uses Apache Kafka more extensively for quueing processing tasks and is capable of exposing the use of Kafka Streams for performing certain analytics.
+
+## Query fusion
+Stroom allows multiple data sources to be queried at the same time and the results of the queries to be fused. This might be for fusing data from multiple search indexes, e.g. events and annotations, or to effectively decorate results with additional data at search time.
+
+## Search result storage
```

Comment:
Great to do this.  But performance might be impacted.
Maybe consider using the disk backed memory store used for reference data for this job too?

Reply:
Yes that is being considered plus fusing data should not impact the immediacy of returned results but instead be performed as a background decoration, i.e. an incremental update.